server-utilities/src/test/java/org/jboss/eap/qe/common/utilities/WaiterTestCase.java

```diff
@@ -0,0 +1,27 @@
+package org.jboss.eap.qe.common.utilities;
+
+import org.jboss.eap.qe.microprofile.common.utilities.Waiter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public class WaiterTestCase {
+
+    @Test
+    public void testWaitingForConditionSucceeds() throws InterruptedException {
+        final Date timeAtStart = new Date();
+
+        final boolean conditionSatisfied = Waiter.waitFor(() -> new Date().getTime() - timeAtStart.getTime() > 4000, 5, TimeUnit.SECONDS);
```

Comment:
shorter timeout pls do the TS runs quickly, what about 4000 => 1200 and 5000 => 2000 ?